src/otx/core/config/__init__.py

```diff
@@ -0,0 +1,31 @@
+"""Config data type objects."""
```

Comment:
Is Structured Configs necessary `src/otx/core/config`? Why not just use yaml configs under `src/otx/config`

Reply:
Maintaining DTO for configuration is super important concept in my slide. Developers must waste their time in the air just to follow what the configuration exactly looks like in their code flow if there is no DTO while working. Please see page 13 at my slides.

Reply:
This appears to be a reference to Hydra's Structured Configs, which offer runtime type checking and compatibility with static type checkers. While these features can be beneficial, I feel they may not be urgently needed at this stage. In DL, conventions like `lr` representing a float for the learning rate are generally well-understood. The requirement to define all Data Transfer Objects (DTOs) at this point might be somewhat excessive.

Reply:
I don't think so. Without this, you might know there is `lr` somewhere in OTX without this. However, you don't get where it exactly is passed in the code flow. I believe that it would take your time >10 mins to find it.

In addition, you can relax the strict structure somewhat relaxed if you need. For example,
https://github.com/openvinotoolkit/training_extensions/blob/8c8c50f64571c4b25ddcf5a1276ebaba5a4bcca9/src/otx/core/config/model.py#L11-L13
You can see that `optimizer` is unstructured one (`dict`).